### DISCUSSION TITLE
Is fast forward working?

### DISCUSSION THREAD

**Discussion by colibris79** [Medard22/Dolphin-MMJR2-VBI#355] [Bug Report]
Hi, first of all thanks for taking the time to give new live to mmjr2
I was using the old discontinued version in my rp3+ till I came across yours. Everything seems to be working great except that fast forward even though it turns on the speed remains at 100%. I know in my rp3+ in the older mmjr2 ff just gets to 130-140%, but it's a good thing to have in games with lots of dialogues, rpgs...

And is there a way to keep both versions installed (mmjr2-revamp and mmjr2-vbi)?

**Comment by Medard22** [comment]
HI there, sorry I missed your message. I wasn´t aware there is a fast forward feature in Dolphin. How can I access it? Setting the speed limit in general settings works . Let me know how can I access fast-forward and I will test it out. 

Unfortunately you cannot keep both revamp and vbi version. For the sake of frontend compatibility (there was already too much of MMJR2 forks) the MMJR2 VBI keeps the same ID as revamp and will overwrite it. It uses a different user folder so it won´t mess up your configurations, saves or shader caches and you can always unistall it and return back to revamp. However they canot be installed both at the same time.

**Reply by Medard22** [reply]
Ah I see, it is pretty hidden. Unfortunately to employ VBI, most of the core timing was rewritten and the throttle option as it was used before was removed. I could probably try to work around it and utilize fast forward by changing speed limit and adding it to the quick menu.

**Reply by Medard22** [reply]
I may have a fix to fast forward toggle, but hotkeys are currently completely broken. I have to find out what went wrong first and after fixing them I will try to reimplement fast forward function. What speed is OK for FF? 150%, 200% or even more?

**Comment by colibris79** [comment]
No worries. Guess it depends on how fast can de rp3+ move it? 200% feels fine where you still can see what's going on

**Reply by Medard22** [reply]
I found out the issue within the Hotkey system and it seems to be working again. I will update the Fast Forward code and cook up a test APK you can try.

**Reply by Medard22** [reply]
Ok, this is the latest dev build based pretty much on Dolphin 5.0-19597 with working fast forward hotkey that toggles between 100% and 200%. Just a reminder, this build bumps UID of shadercache due to changes to graphics pipeline, so there might be shader compilation stuttering present even on games that had shaders cached. I will probably make a hotfix to lastest release APK. So you can wait a while and update with updated APK from releases section.

Thanks for letting me know! I´ve never used Hotkeys and I would probably never find out they are broken.

https://drive.google.com/file/d/1jWFeVeZjimAHRodSzxuDdlMxZhFOq2N5/view?usp=sharing

**Reply by colibris79** [reply]
Thank you for taking the time to improve the app!!! I'll give it a try and let you know how it goes

**Reply by lezzob** [reply]
Hello! Sorry to dig up this very old thread but I really wanted to know if is there any way to have this feature included again. Like a hotkey to disable the speed limit of emulation allowing the cpu to go to 200%/unlimited speed without going through the menus. I tried to download this version here but it seems the file was deleted from GDrive. Thank you

**Comment by colibris79** [comment]
Working great, @Medard22 
Tried with Paper Mario (reached 150-160% fast forward speed) and Luigi's Mansion (around 180%). 
So thanks for implementing it back!
